src/openrct2/peep/Guest.cpp

```diff
@@ -7373,11 +7375,6 @@ static bool IsThoughtShopItemRelated(const PeepThoughtType type)
 
 void Guest::RemoveRideFromMemory(ride_id_t rideId)
 {
-    uint8_t ride_id_bit = rideId % 8;
-    uint8_t ride_id_offset = rideId / 8;
-
-    // clear ride from potentially being in RidesBeenOn
-    RidesBeenOn[ride_id_offset] &= ~(1 << ride_id_bit);
```

Comment:
Shouldn't this have been replaced with a call to remove it from the ride use history?

Reply:
It's handled differently. At this point it has already been removed from the ride history.